packages/docusaurus-utils-validation/src/tags.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+import fs from 'fs-extra';
+import _ from 'lodash';
+import Joi from 'joi';
+import YAML from 'js-yaml';
+import type {
+  Tag,
+  TagsFile,
+  TagsFileInput,
+  TagsPluginOptions,
+} from '@docusaurus/utils';
+
+const tagDefinitionSchema = Joi.object<TagsFile>().pattern(
+  Joi.string(),
+  Joi.object({
+    label: Joi.string(),
+    description: Joi.string(),
+    permalink: Joi.string(),
+  }).allow(null),
+);
+
+function validateDefinedTags(tags: unknown): Joi.ValidationResult<TagsFile> {
+  return tagDefinitionSchema.validate(tags);
+}
```

Comment:
```suggestion
const tagDefinitionSchema = Joi.object<TagsFileInput>().pattern(
  Joi.string(),
  Joi.object({
    label: Joi.string(),
    description: Joi.string(),
    permalink: Joi.string(),
  }).allow(null),
);

function validateDefinedTags(tags: unknown): Joi.ValidationResult<TagsFileInput> {
  return tagDefinitionSchema.validate(tags);
}
```

What we get and validate from the user might still contain undefined label/permalink